packages/anthropic-vertex/src/anthropic-messages-language-model.ts

```diff
@@ -0,0 +1,528 @@
+import {
+  LanguageModelV1,
+  LanguageModelV1CallWarning,
+  LanguageModelV1FinishReason,
+  LanguageModelV1FunctionToolCall,
+  LanguageModelV1StreamPart,
+  UnsupportedFunctionalityError,
+} from '@ai-sdk/provider';
+import {
+  ParseResult,
+  combineHeaders,
+  createEventSourceResponseHandler,
+  createJsonResponseHandler,
+  postJsonToApi,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import { anthropicFailedResponseHandler } from './anthropic-error';
+import {
+  AnthropicMessagesModelId,
+  AnthropicMessagesSettings,
+} from './anthropic-messages-settings';
+import { convertToAnthropicMessagesPrompt } from './convert-to-anthropic-messages-prompt';
+import { mapAnthropicStopReason } from './map-anthropic-stop-reason';
+import { GoogleAuth } from 'google-auth-library';
+
+type AnthropicMessagesConfig = {
+  provider: string;
+  baseURL: string;
+  headers: () => Record<string, string | undefined>;
+  fetch?: typeof fetch;
+  projectId?: string;
+  region?: string;
+  googleAuth?: GoogleAuth;
+};
+
+export class AnthropicMessagesLanguageModel implements LanguageModelV1 {
+  readonly specificationVersion = 'v1';
+  readonly defaultObjectGenerationMode = 'tool';
+  readonly supportsImageUrls = false;
+  readonly defaultVersion = 'vertex-2023-10-16';
+
+  readonly modelId: AnthropicMessagesModelId;
+  readonly settings: AnthropicMessagesSettings;
+  readonly path: string;
+
+  private readonly config: AnthropicMessagesConfig;
+
+  constructor(
+    modelId: AnthropicMessagesModelId,
+    settings: AnthropicMessagesSettings,
+    config: AnthropicMessagesConfig,
+  ) {
+    this.modelId = modelId;
+    this.settings = settings;
+    this.config = config;
+    this.path = `/projects/${config.projectId}/locations/${config.region}/publishers/anthropic/models/${modelId}`;
```

Comment:
For some reason I had to remove the starting `/` from this string to get this provider to work. Otherwise I get an error:
```
13 |     statusCode === 429 || // too many requests
14 |     statusCode >= 500),
15 |     // server error
16 |     data
17 |   }) {
18 |     super(message);
         ^
AI_APICallError: Not Found
 url: "https://europe-west1-aiplatform.googleapis.com/v1//projects/[project id]/locations/europe-west1/publishers/anthropic/models/claude-3-5-sonnet@20240620:rawPredict"

      at new APICallError (.../ai/node_modules/.pnpm/@ai-sdk+provider@0.0.14/node_modules/@ai-sdk/provider/dist/index.mjs:18:5)
      at ../ai/node_modules/.pnpm/@ai-sdk+provider-utils@1.0.5_zod@3.23.8/node_modules/@ai-sdk/provider-utils/dist/index.mjs:458:15
```
Note the double slash after the API base URL. I have no clue there this additional slash comes from and this was the only place I knew where I could get rid of it. It is most certainly the wrong place.